src/Extractor/BaseExtractor.php

```diff
@@ -76,25 +69,30 @@ public function __construct(array $parameters, array $state, LoggerInterface $lo
         }
     }
 
+    abstract public function testConnection(): void;
+
+    abstract public function simpleQuery(ExportConfig $exportConfig): string;
+
     /**
      * @return PDO|mixed
      */
-    abstract public function createConnection(DatabaseConfig $databaseConfig);
-
-    abstract public function testConnection(): void;
+    abstract protected function createConnection(DatabaseConfig $databaseConfig);
 
-    abstract public function simpleQuery(ExportConfig $exportConfig): string;
+    abstract protected function getMetadataProvider(): MetadataProvider;
 
-    abstract public function getMaxOfIncrementalFetchingColumn(ExportConfig $exportConfig): ?string;
+    abstract protected function getMaxOfIncrementalFetchingColumn(ExportConfig $exportConfig): ?string;
```

Comment:
Internal methods should be protected, ... it doesn't limit from making it public - in some extractor - if you really need it.